_playwright-tests/UI/CustomRepoCRUD.spec.ts

```diff
@@ -0,0 +1,97 @@
+export const repoName = `${repoNamePrefix}-${randomName()}`;
+export const rank = () => Math.floor(Math.random() * 10 + 1).toString();
+export const randomNum = () =>
+  Math.floor(Math.random() * 10 + 1)
+    .toString()
+    .padStart(2, '0');
+
+export const url = `https://stephenw.fedorapeople.org/multirepos/${rank()}/repo${randomNum()}/`;
+
+test.describe('Custom Repositories CRUD', () => {
+  test('Add, Read, update, delete a repo', async ({ page }) => {
+    await test.step('Delete any CRUD test repos that exist', async () => {
+      await navigateToRepositories(page);
+      await closePopupsIfExist(page);
+      await deleteAllRepos(page, `&search=${repoNamePrefix}`);
+    });
+    await test.step('Create a repository', async () => {
+      // Click on the 'Add repositories' button
+      await navigateToRepositories(page);
+      await closePopupsIfExist(page);
+      await page.getByRole('button', { name: 'Add repositories' }).first().click();
+      const repositoryModal = page.locator('div[id^="pf-modal-part"]').first();
+      await expect(repositoryModal).toBeVisible();
+      await expect(page.locator('text="Add custom repositories"')).toBeVisible();
+      // Fill in the repository details
+      await page.getByLabel('Name').fill(`${repoName}`);
+      await page.getByLabel('Introspect only').click();
+      await page.getByLabel('URL').fill(url);
+      await page.getByRole('button', { name: 'Save' }).first().click();
+    });
+    await test.step('Wait for status to be "Valid"', async () => {
+      const row = await getRowByNameOrUrl(page, repoName);
+      await expect(row.getByText('Valid')).toBeVisible({ timeout: 60000 });
+    });
+    await test.step('Read the repo', async () => {
+      // Search for the created repo
+      await page.getByRole('textbox', { name: 'Filter by name/url' }).fill(repoName);
+      await expect(page.getByRole('row', { name: `${repoName}` })).toBeVisible();
+      if (await page.getByLabel('Kebab toggle').first().isDisabled())
+        throw Error("Kebab is disabled when it really shouldn't be");
+      await page.getByLabel('Kebab toggle').first().click();
+      // Click on the Edit button to see the repo
+      await page.getByRole('menuitem', { name: 'Edit' }).click();
+      await expect(page.getByText('Edit custom repository')).toBeVisible();
+      // Assert we can read some values
+      await expect(page.getByTestId('input_name')).toBeVisible();
+      const inputValue = await page.getByTestId('input_name').inputValue();
+      await expect(inputValue).toBe(`${repoName}`);
+      const inputValue2 = await page.getByTestId('input_url').inputValue();
+      expect(inputValue2).toBe(`${url}`);
+    });
+    await test.step('Update the repository', async () => {
+      await page.getByTestId('input_name').fill(`${repoName}-Edited`);
+      await page.getByLabel('Snapshotting').click();
+      await page.getByRole('button', { name: 'Save' }).first().click();
```

Comment:
Same questions as above - why using TestId and if there are multiple Save buttons.

Reply:
I changed to:
+      await page.getByRole('button', { name: 'Save', exact: true }).click();